crates/octez/src/async/rollup.rs

```diff
@@ -0,0 +1,228 @@
+        self.rpc_endpoint = Some(rpc_endpoint.clone());
+        self
+    }
+
+    pub fn build(self) -> Result<OctezRollupConfig> {
+        Ok(OctezRollupConfig {
+            binary_path: self
+                .binary_path
+                .unwrap_or(PathBuf::from(DEFAULT_BINARY_PATH)),
+            octez_client_base_dir: self.octez_client_base_dir,
+            octez_node_endpoint: self.octez_node_endpoint,
+            data_dir: self.data_dir,
+            address: self.address,
+            operator: self.operator,
+            boot_sector_file: self.boot_sector_file,
+            rpc_endpoint: self.rpc_endpoint.unwrap_or_else(|| {
+                let uri = Uri::from_str(&format!("127.0.0.1:{}", unused_port())).unwrap();
+                Endpoint::try_from(uri).unwrap()
+            }),
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct OctezRollupConfig {
+    pub binary_path: PathBuf,
+    pub octez_client_base_dir: PathBuf,
+    pub octez_node_endpoint: Endpoint,
+    pub data_dir: RollupDataDir,
+    pub address: SmartRollupHash,
+    pub operator: String,
+    pub boot_sector_file: PathBuf,
+    pub rpc_endpoint: Endpoint,
+}
+
+pub struct OctezRollup {
+    binary_path: PathBuf,
+    /// Path to the directory where the rollup state & kernel preimages are stored
+    data_dir: PathBuf,
+    octez_client_base_dir: PathBuf,
+    octez_node_endpoint: Endpoint,
+    rpc_endpoint: Endpoint,
+}
+
+impl OctezRollup {
+    pub fn new(
+        binary_path: &Path,
+        data_dir: &Path,
+        octez_client_base_dir: &Path,
+        octez_node_endpoint: &Endpoint,
+        rpc_endpoint: &Endpoint,
+    ) -> Self {
+        Self {
+            binary_path: binary_path.to_path_buf(),
+            data_dir: data_dir.to_path_buf(),
+            octez_client_base_dir: octez_client_base_dir.to_path_buf(),
+            octez_node_endpoint: octez_node_endpoint.clone(),
+            rpc_endpoint: rpc_endpoint.clone(),
+        }
+    }
+}
+
+impl OctezRollup {
+    fn command(&self) -> Command {
+        let mut command = Command::new(self.binary_path.to_string_lossy().as_ref());
+        command.args([
+            "--endpoint",
+            &self.octez_node_endpoint.to_string(),
+            "--base-dir",
+            &self.octez_client_base_dir.to_string_lossy(),
+        ]);
+        command
+    }
+
+    pub fn run(
+        &self,
+        address: &SmartRollupHash,
+        operator: &str,
+        boot_sector_file: &Path,
+    ) -> Result<Child> {
+        let mut command = self.command();
+        command.args([
+            "run",
+            "operator",
+            "for",
+            &address.to_string(),
+            "with",
+            "operators",
+            operator,
+            "--data-dir",
+            &self.data_dir.to_string_lossy(),
+            "--rpc-addr",
+            self.rpc_endpoint.host(),
+            "--rpc-port",
+            &self.rpc_endpoint.port().to_string(),
+            "--boot-sector-file",
```

Comment:
I think `--boot-sector-file` should be optional since it's necessary only for bootstrap rollups

Reply:
in our usecase we are only using bootstrap jstz rollup IIUC so i think it's better to make it required for simplicity. 
cc @zcabter

Reply:
actually this is in octez so we can keep it optional, fixed